src/testrel.jl

```diff
@@ -633,17 +638,25 @@ function _test_rel_step(
         #   If an abort is expected it is encountered
         #   If no abort is expected it is not encountered
         #   If results are expected, they are found (other results are ignored)
-        #   If problems are expected, they are found (other problems are ignored)
-        #   If no problems are expected, warning level problems are ignored
+        #   If problems are expected, they are found
+        #   Unexpected problems with severity worse than allowable are not found
 
         unexpected_errors_found = false
+        error_levels = []
+        if step.allow_unexpected == :warning
+            push!(error_levels, "error")
+            push!(error_levels, "exception")
```

Comment:
exception should always be in the set, right?

Reply:
I'm not sure... There is a significant difference internally, but by the time we reach here the two are very similar. It does make sense that an exception should never be an acceptable, expected, behaviour, though.

Reply:
I made the levels more correct, explicit, and now exception is always in the set.